apis/python/tests/test_dataframe.py

```diff
@@ -651,7 +665,31 @@ def test_extend_enumeration_values_deduplication(tmp_path, deduplicate, ordered)
             "float32_enum": pa.array([2.25, 3.75], type=pa.float32()),
             "bool_enum": pa.array([True], type=pa.bool_()),
         }
-        sdf.extend_enumeration_values(values)
+
+        if extend_not_first_write:
+            sdf.extend_enumeration_values(values)
+        else:
+            data = pa.Table.from_pydict(
+                {
+                    "soma_joinid": pa.array([0, 1, 2, 3], type=pa.int64()),
+                    "not_an_enum": pa.array(
+                        ["the", "quick", "brown", "fox"], type=pa.large_string()
+                    ),
+                    "string_enum": pa.DictionaryArray.from_arrays(
+                        [0, 1, 0, 1], values["string_enum"]
+                    ),
+                    "int64_enum": pa.DictionaryArray.from_arrays(
+                        [0, 2, 3, 1], values["int64_enum"]
+                    ),
+                    "float32_enum": pa.DictionaryArray.from_arrays(
+                        [0, 0, 0, 1], values["float32_enum"]
```

Comment:
This needs to also verify that NaN's are handled correctly, as that is tricky (especially given that the different Nan variants should not de-dup).

I suggest something like [NaN, -NaN , a signalling NaN, etc].  There is code to generate most of these in `test_enum_handling_category_of_nan_62449`

Reply:
@bkmartinjr 
https://github.com/single-cell-data/TileDB-SOMA/blob/kerl/enum-values-2/apis/python/tests/test_dataframe.py#L735-L753

Reply:
(i.e. please scroll down)

Reply:
where does it check that the expected extension occurred?   I.e., read back and assert that the values are as expected?

(really sorry if I'm being blind!)

Reply:
https://github.com/single-cell-data/TileDB-SOMA/pull/3815/commits/8cf48c59e279564effa105aff3a0315328c00ae3